### ISSUE TITLE
DNS is not working with coreos/gce

### ISSUE THREAD

**Issue by yifan-gu** [kubernetes/kubernetes#15591]
Trying to run e2e tests with docker/coreos/gce, but failed as the [kibana-logging](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/kibana-controller.yaml) pod keeps failing.
```
kubectl get pod --all-namespaces
NAMESPACE     NAME                                   READY     STATUS             RESTARTS   AGE
kube-system   elasticsearch-logging-v1-0lpmt         1/1       Running            0          1h
kube-system   elasticsearch-logging-v1-qrd3p         1/1       Running            0          1h
kube-system   heapster-v10-dbqaz                     1/1       Running            4          1h
kube-system   kibana-logging-v1-6y6nq                0/1       CrashLoopBackOff   27         1h
kube-system   kube-dns-v9-zmsw9                      4/4       Running            0          1h
kube-system   kube-ui-v3-da42d                       1/1       Running            0          1h
kube-system   monitoring-influxdb-grafana-v2-6hwc8   2/2       Running            0          1h
```

From the docker container's [log](https://gist.github.com/yifan-gu/6db2d99d251370973970), it seems like a the pod cannot reach `http://elasticsearch-logging:9200`

I also did `nslookup` within a pod according to the [dns doc](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/dns#how-do-i-test-if-it-is-working), it fails as well:

```
kubectl exec busybox -- nslookup kubernetes.default
Server:    10.0.0.10
Address 1: 10.0.0.10

nslookup: can't resolve 'kubernetes.default'
```

**Comment by ArtfulCoder** [comment]
Can you try kubectl exec busybox -- nslookup kubernetes.default.svc.cluster.local

**Comment by ArtfulCoder** [comment]
@yifan-gu are you still facing a dns issue ?

**Comment by yifan-gu** [comment]
> @yifan-gu are you still facing a dns issue ?

Yes. 
```
kubectl exec busybox -- nslookup kubernetes.default.svc.cluster.local
Server:    10.0.0.10
Address 1: 10.0.0.10

nslookup: can't resolve 'kubernetes.default.svc.cluster.local'
```

**Comment by thockin** [comment]
Is the DNS service actually at `10.0.0.10` ?

Is the cluster domain actually set to `cluster.local` ?

Is kube-proxy actually running?

If you delete teh DNS pod and let it recreate does it still fail?

**Comment by yifan-gu** [comment]
Fixed by #15749
